lib/src/track/visualizer_web.dart

```diff
@@ -0,0 +1,168 @@
+import 'dart:async';
+import 'dart:js_interop';
+import 'dart:math' as math;
+import 'dart:typed_data';
+
+import 'package:flutter_webrtc/flutter_webrtc.dart';
+
+import 'package:livekit_client/src/events.dart' show AudioVisualizerEvent;
+import '../logger.dart' show logger;
+import 'local/local.dart' show AudioTrack;
+import 'visualizer.dart';
+import 'web/_audio_analyser.dart';
+
+class VisualizerWeb extends Visualizer {
+  AudioAnalyser? _audioAnalyser;
+  MultiBandTrackVolumeOptions options = MultiBandTrackVolumeOptions();
+  Timer? _timer;
+  final AudioTrack? _audioTrack;
+  MediaStreamTrack get mediaStreamTrack => _audioTrack!.mediaStreamTrack;
+
+  final VisualizerOptions visualizerOptions;
+
+  VisualizerWeb(this._audioTrack, {required this.visualizerOptions}) {
+    onDispose(() async {
+      await events.dispose();
+    });
+  }
+
+  @override
+  Future<void> start() async {
+    if (_audioAnalyser != null) {
+      return;
+    }
+
+    final bands = visualizerOptions.barCount;
+
+    _audioAnalyser = createAudioAnalyser(_audioTrack!, options.analyserOptions);
+
+    final bufferLength = _audioAnalyser?.analyser.frequencyBinCount;
+
+    _timer = Timer.periodic(
+      Duration(milliseconds: options.updateInterval!.toInt()),
+      (timer) {
+        try {
+          var tmp = JSFloat32Array.withLength(bufferLength ?? 0);
+          _audioAnalyser?.analyser.getFloatFrequencyData(tmp);
+          Float32List frequencies = Float32List(tmp.toDart.length);
+          for (var i = 0; i < tmp.toDart.length; i++) {
+            var element = tmp.toDart[i];
+            frequencies[i] = element;
+          }
+          frequencies = frequencies.sublist(
+              options.loPass!.toInt(), options.hiPass!.toInt());
+
+          final normalizedFrequencies =
+              normalizeFrequencies(frequencies); // is this needed ?
+          final chunkSize = (normalizedFrequencies.length / (bands + 1))
+              .ceil(); // we want logarithmic chunking here
```

Comment:
the comment here was intended of more as a `TODO` so that the bins would more accurately reflect how humans hear. I think we can remove the comment for as long as we don't have that.